core/src/main/java/io/confluent/kafka/schemaregistry/storage/InMemoryCache.java

```diff
@@ -38,19 +38,17 @@
 public class InMemoryCache<K, V> implements LookupCache<K, V> {
   // visible for subclasses
   protected final ConcurrentNavigableMap<K, V> store;
-  private final Map<Integer, SchemaKey> guidToSchemaKey;
-  private final Map<MD5, SchemaIdAndSubjects> schemaHashToGuid;
-  private final Map<Integer, List<SchemaKey>> guidToDeletedSchemaKeys;
+  private final Map<Integer, Map<String, Integer>> guidToSubjectVersions;
+  private final Map<MD5, Integer> hashToGuid;
 
```

Comment:
I've simplified the internal maps. 

Before we had `(guid -> schemakey)` and `(md5 -> (guid -> (subject -> version))`

I've refactored these to `(guid -> (subject -> version))` and `(md5 -> guid)`

The `(guid -> schemaKey)` can be reconstructed by taking the first entry from `(guid -> (subject -> version))`

Also we no longer need `guidToDeletedSchemaKeys` because tombstoning now happens during registration as opposed to in `KafkaStoreMessageHandler`